src/audiovideocontroller/DefaultAudioVideoController.ts

```diff
@@ -752,6 +752,11 @@ export default class DefaultAudioVideoController
 
   /* @internal */
   stopReturningPromise(): Promise<void> {
+    // We need to ensure this disables reconnection before any other steps are taken,
+    // in case something during close is triggering a reconnection (e.g. websocket
+    // close)
+    this._reconnectController.disableReconnect();
```

Comment:
Just bringing this change over from previous commit. It logically should be this way to avoid race conditions.